File: app/src/main/java/com/mucheng/nodejava/MainActivity.java
import android.app.Activity;
import android.os.Bundle;
import com.mucheng.nodejava.core.Context;
import com.mucheng.nodejava.core.Isolate;

import java.io.IOException;
import java.io.InputStream;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Isolate isolate = new Isolate();
        Context context = new Context(isolate);
        context.injectJavaBridge();
        context.evaluateScript(readFromAssets("nodeJava.js"));
        context.evaluateScript(readFromAssets("test.js"));
    }

    private String readFromAssets(String path) {
        String result = "";
        try {
            InputStream is = getAssets().open(path);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            result = new String(buffer, "UTF-8");
        } catch (IOException e) {
            e.printStackTrace();
        }
        return result;
    }
}

File: app/src/main/java/com/mucheng/nodejava/MainActivity.kt
package com.mucheng.nodejava

import android.app.Activity
import android.os.Bundle
import com.mucheng.nodejava.core.Context
import com.mucheng.nodejava.core.Isolate
import kotlin.concurrent.thread

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val isolate = Isolate()
        val context = Context(isolate)

        thread {
            Thread.sleep(1000L)
            context.emitProcessExit()
        }

        context.evaluateScript("setInterval(() => console.log('ticked'), 200);")
        context.spinEventLoop()

        context.evaluateScript("console.log('exec done')")
    }

}
